app/app.go

```diff
@@ -1741,11 +1765,16 @@ func (d *Deployments) updateDeviceDeploymentsStatus(
 				return err
 			}
 			if ok {
-				_, err := d.createDeviceDeploymentWithStatus(ctx,
+				deviceDeployment, err := d.createDeviceDeploymentWithStatus(ctx,
 					deviceId, deployment, status)
 				if err != nil {
 					return err
 				}
+				if err := d.reindexDeployment(ctx, deviceDeployment.DeviceId,
+					deviceDeployment.DeploymentId, deviceDeployment.Id); err != nil {
+					l := log.FromContext(ctx)
+					l.Warn(errors.Wrap(err, "failed to trigger a device reindex"))
+				}
```

Comment:
Shouldn't there be a condition on the status being a terminal state (`success`, `failure`, `already-installed`, `noartifact`, `decommissioned` or `aborted`)?

Reply:
you are right